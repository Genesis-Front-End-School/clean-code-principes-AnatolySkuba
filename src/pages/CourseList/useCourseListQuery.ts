import { useQuery } from "react-query";
import { getAllCourses } from "../../services/API";
import { ROUTER_KEYS } from "../../utils/constants";

export const useCourseListQuery = () => {
    const { data, isLoading, isError } = useQuery(`${ROUTER_KEYS.COURSES}`, getAllCourses);
    return { data, isLoading, isError };
};
