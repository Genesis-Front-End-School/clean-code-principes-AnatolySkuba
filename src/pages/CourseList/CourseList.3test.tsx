import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { CourseList } from './CourseList';
import { useCourseListQuery } from './useCourseListQuery';
import { renderWithRouter } from '../../tests/helpers/renderWithRouter';

const mockedUseCourseListQuery = useCourseListQuery as jest.Mock;
jest.mock('./useCourseListQuery');

jest.mock('hls.js', () => {
  class Hls {
    loadSource() {}
    attachMedia() {}
  }
  return Hls;
});

jest.mock('../../components/Loader', () => ({
  __esModule: true,
  Loader: () => 'Mocked the Loader Component',
}));

jest.mock('../../components/Error', () => ({
  __esModule: true,
  Error: () => 'Mocked the Error Component',
}));

jest.mock('../../components/CourseCard', () => ({
  __esModule: true,
  CourseCard: () => 'Mocked the CourseCard Component',
}));

jest.mock('../../components/Pagination', () => ({
  __esModule: true,
  Pagination: () => 'Mocked the Pagination Component',
}));

const MockData = [
  { id: 1, title: 'Course 1' },
  { id: 2, title: 'Course 2' },
  { id: 3, title: 'Course 3' },
  { id: 4, title: 'Course 4' },
  { id: 5, title: 'Course 5' },
  { id: 6, title: 'Course 6' },
  { id: 7, title: 'Course 7' },
  { id: 8, title: 'Course 8' },
  { id: 9, title: 'Course 9' },
  { id: 10, title: 'Course 10' },
  { id: 11, title: 'Course 11' },
];

describe('CourseList component', () => {
  afterEach(jest.clearAllMocks);

  it('Displays the loading view', () => {
    mockedUseCourseListQuery.mockImplementation(() => ({
      isLoading: true,
    }));

    renderWithRouter(<CourseList />);

    expect(screen.getByText(/Mocked the Loader Component/)).toBeTruthy();
  });

  it('Displays the Error Component', () => {
    mockedUseCourseListQuery.mockImplementation(() => ({
      isError: true,
    }));

    renderWithRouter(<CourseList />);

    console.log(83, screen.debug());

    expect(screen.getByText(/Mocked the Error Component/)).toBeTruthy();
  });

  it('Render CourseList correctly', () => {
    mockedUseCourseListQuery.mockImplementation(() => ({
      courses: MockData,
    }));

    renderWithRouter(<CourseList />);
    console.log(91, screen.debug());

    expect(screen.getByRole('listbox')).toBeInTheDocument();
  });
});
