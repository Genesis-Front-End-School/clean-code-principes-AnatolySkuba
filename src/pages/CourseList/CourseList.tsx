import React from "react";
import { useSearchParams } from "react-router-dom";

import { CourseCard } from "../../components";
import { Course } from "../../utils/types";
import { useCourseListQuery } from "./useCourseListQuery";

export const CourseList = () => {
    const [searchParams, setSearchParams] = useSearchParams();
    const { data, isLoading, isError } = useCourseListQuery();

    const page = Number(searchParams.get("page")) || 1;

    if (isLoading) {
        return <div data-testid="courses-loading">Loading...</div>;
    }

    if (isError) {
        return <div data-testid="courses-error">Error!</div>;
    }

    const pages = Math.ceil(data?.data.courses.length / 10);
    const courses = data?.data.courses.slice((page - 1) * 10, page * 10);

    return (
        <>
            <ul
                className="grid gap-5 p-4 justify-items-center tablet:grid-cols-2 laptop:grid-cols-3 desktop:grid-cols-4"
                role="listbox"
            >
                {courses?.map((course: Course) => (
                    <CourseCard key={course.id} {...course} />
                ))}
            </ul>

            {/* Pagination */}
            <nav
                aria-label="Pagination"
                className="flex justify-center items-center text-gray-600 mt-8 lg:mt-0 caret-transparent"
            >
                <p
                    className="p-2 ml-4 rounded hover:bg-gray-100 cursor-pointer"
                    onClick={() =>
                        setSearchParams({
                            page: (page > 1 ? page - 1 : page).toString(),
                        })
                    }
                >
                    <svg
                        xmlns="http://www.w3.org/2000/svg"
                        className="h-6 w-6"
                        fill="none"
                        viewBox="0 0 24 24"
                        stroke="currentColor"
                    >
                        <path
                            strokeLinecap="round"
                            strokeLinejoin="round"
                            strokeWidth="2"
                            d="M15 19l-7-7 7-7"
                        />
                    </svg>
                </p>
                <p
                    className={`w-10 h-10 text-center leading-10 rounded hover:bg-gray-100 cursor-pointer ${
                        page === 1 && "bg-gray-200 text-gray-900 font-medium"
                    }`}
                    onClick={() => setSearchParams({ page: "1" })}
                >
                    1
                </p>
                <p
                    className={`w-10 h-10 text-center leading-10 rounded hover:bg-gray-100 cursor-pointer ${
                        page === 2 && "bg-gray-200 text-gray-900 font-medium"
                    }`}
                    onClick={() => setSearchParams({ page: "2" })}
                >
                    2
                </p>
                <p
                    className={`w-10 h-10 text-center leading-10 rounded hover:bg-gray-100 cursor-pointer ${
                        page === 3 && "bg-gray-200 text-gray-900 font-medium"
                    }`}
                    onClick={() => setSearchParams({ page: "3" })}
                >
                    3
                </p>
                <p
                    className="p-2 ml-4 rounded hover:bg-gray-100 cursor-pointer"
                    onClick={() =>
                        setSearchParams({
                            page: (pages > page ? page + 1 : page).toString(),
                        })
                    }
                >
                    <svg
                        xmlns="http://www.w3.org/2000/svg"
                        className="h-6 w-6"
                        fill="none"
                        viewBox="0 0 24 24"
                        stroke="currentColor"
                    >
                        <path
                            strokeLinecap="round"
                            strokeLinejoin="round"
                            strokeWidth="2"
                            d="M9 5l7 7-7 7"
                        />
                    </svg>
                </p>
            </nav>
        </>
    );
};
