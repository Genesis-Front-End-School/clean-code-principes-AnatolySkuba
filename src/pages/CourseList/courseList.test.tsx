import { screen } from "@testing-library/react";

import { CourseList } from "./CourseList";
import { useCourseListQuery } from "./useCourseListQuery";
import { renderWithRouter } from "../../tests/helpers/renderWithRouter";

const mockedUseCourseListQuery = useCourseListQuery as jest.Mock;
jest.mock("./useCourseListQuery");

const response = {
    data: {
        courses: [
            {
                containsLockedLessons: true,
                description:
                    "Reignite your inner drive by managing factors that dampen your motivation.",
                duration: 521,
                id: "352be3c6-848b-4c19-9e7d-54fe68fef183",
                launchDate: "2023-03-06T16:50:06.000Z",
                lessonsCount: 2,
                meta: {
                    slug: "lack-of-motivation-how-to-overcome-it",
                    skills: [
                        "Aligning your goals with your motives",
                        "Overcoming decision paralysis",
                        "Reframing negative self-talk",
                        "Gaining clarity",
                        "Challenging yourself",
                    ],
                    courseVideoPreview: {
                        link: "https://wisey.app/videos/lack-of-motivation-how-to-overcome-it/preview/AppleHLS1/preview.m3u8",
                        duration: 27,
                        previewImageLink:
                            "https://wisey.app/assets/images/web/course-covers/lack-of-motivation-how-to-overcome-it/preview",
                    },
                },
                previewImageLink:
                    "https://wisey.app/assets/images/web/course-covers/lack-of-motivation-how-to-overcome-it",
                rating: 3.5,
                status: "launched",
                tags: ["productivity"],
                title: "Lack of Motivation & How to Overcome It",
            },
        ],
    },
};

describe("CourseList component", () => {
    afterEach(jest.clearAllMocks);

    test("Displays the loading view", () => {
        mockedUseCourseListQuery.mockImplementation(() => ({
            isLoading: true,
        }));
        renderWithRouter(<CourseList />);
        expect(screen.getByTestId("courses-loading")).toBeInTheDocument();
        expect(screen.getByText(/Loading.../i)).toBeVisible();
    });

    test("Displays the error message", () => {
        mockedUseCourseListQuery.mockImplementation(() => ({
            isError: true,
        }));
        renderWithRouter(<CourseList />);
        expect(screen.getByTestId("courses-error")).toBeInTheDocument();
        expect(screen.getByText(/Error!/i)).toBeVisible();
    });

    test("Render CourseList correctly", () => {
        mockedUseCourseListQuery.mockImplementation(() => ({
            data: response,
        }));
        renderWithRouter(<CourseList />);
        expect(screen.getByRole("listbox")).toBeInTheDocument();
        expect(screen.getByRole("navigation")).toBeInTheDocument();
    });

    test("Queries are called", () => {
        mockedUseCourseListQuery.mockImplementation(() => ({
            data: response,
        }));
        renderWithRouter(<CourseList />);
        expect(mockedUseCourseListQuery).toBeCalledTimes(1);
    });

    test("CourseList snapshots", () => {
        mockedUseCourseListQuery.mockImplementation(() => ({
            data: response,
        }));
        renderWithRouter(<CourseList />);
        expect(screen.getByRole("listbox")).toMatchSnapshot();
        expect(screen.getByRole("navigation")).toMatchSnapshot();
    });
});
