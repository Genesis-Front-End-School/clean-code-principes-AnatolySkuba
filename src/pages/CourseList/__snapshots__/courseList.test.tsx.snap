// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseList component CourseList snapshots 1`] = `
<ul
  class="grid gap-5 p-4 justify-items-center tablet:grid-cols-2 laptop:grid-cols-3 desktop:grid-cols-4"
  role="listbox"
>
  <li
    class="block min-w-80 rounded-lg p-3 border shadow-md shadow-gray-500 hover:scale-105"
  >
    <a
      href="course/352be3c6-848b-4c19-9e7d-54fe68fef183"
    >
      <div
        class="relative h-28 w-64 mx-auto group"
        data-testid="group"
      >
        <img
          alt="Lack of Motivation & How to Overcome It"
          class="absolute top-0 h-full w-full rounded-md object-cover group-hover:hidden"
          src="https://wisey.app/assets/images/web/course-covers/lack-of-motivation-how-to-overcome-it/cover.webp"
        />
        <video
          class="absolute top-0 h-full w-full opacity-0 rounded-md object-cover group-hover:opacity-100"
          data-testid="video-player"
          id="352be3c6-848b-4c19-9e7d-54fe68fef183"
          loop=""
          preload="none"
        />
      </div>
      <div
        class="mt-1"
      >
        <p
          class="font-medium"
        >
          Lack of Motivation & How to Overcome It
        </p>
        <p
          class="text-sm text-gray-500"
        >
          Reignite your inner drive by managing factors that dampen your motivation.
        </p>
        <div
          class="flex items-center gap-3 mt-1 text-xs"
        >
          <div>
            <p
              class="text-gray-500"
            >
              Lessons
            </p>
            <p
              class="flex font-medium gap-1"
            >
              <svg
                enable-background="new 0 0 48 48"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                version="1"
                viewBox="0 0 48 48"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="#D1C4E9"
                >
                  <path
                    d="M38,7H10C8.9,7,8,7.9,8,9v6c0,1.1,0.9,2,2,2h28c1.1,0,2-0.9,2-2V9C40,7.9,39.1,7,38,7z"
                  />
                  <path
                    d="M38,19H10c-1.1,0-2,0.9-2,2v6c0,1.1,0.9,2,2,2h28c1.1,0,2-0.9,2-2v-6C40,19.9,39.1,19,38,19z"
                  />
                  <path
                    d="M38,31H10c-1.1,0-2,0.9-2,2v6c0,1.1,0.9,2,2,2h28c1.1,0,2-0.9,2-2v-6C40,31.9,39.1,31,38,31z"
                  />
                </g>
                <circle
                  cx="38"
                  cy="38"
                  fill="#43A047"
                  r="10"
                />
                <polygon
                  fill="#DCEDC8"
                  points="42.5,33.3 36.8,39 34.1,36.3 32,38.5 36.8,43.3 44.6,35.5"
                />
              </svg>
              2
            </p>
          </div>
          <div>
            <p
              class="text-gray-500"
            >
              Rating
            </p>
            <p
              class="flex font-medium gap-1"
            >
              <svg
                enable-background="new 0 0 48 48"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                version="1"
                viewBox="0 0 48 48"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="#00BCD4"
                >
                  <rect
                    height="20"
                    width="10"
                    x="19"
                    y="22"
                  />
                  <rect
                    height="34"
                    width="10"
                    x="32"
                    y="8"
                  />
                  <rect
                    height="12"
                    width="10"
                    x="6"
                    y="30"
                  />
                </g>
                <g
                  fill="#3F51B5"
                >
                  <polygon
                    points="11,8 21,18 21,8"
                  />
                  <rect
                    height="14.1"
                    transform="matrix(-.707 -.707 .707 -.707 10.879 36.506)"
                    width="4"
                    x="11"
                    y="8.9"
                  />
                </g>
              </svg>
              3.5
            </p>
          </div>
          <div>
            <div
              class="flex gap-2"
            >
              <p
                class="text-gray-500"
              >
                Skills
              </p>
              <svg
                enable-background="new 0 0 48 48"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                version="1"
                viewBox="0 0 48 48"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M40,22h-8l-4,4V12c0-2.2,1.8-4,4-4h8c2.2,0,4,1.8,4,4v6C44,20.2,42.2,22,40,22z"
                  fill="#2196F3"
                />
                <circle
                  cx="17"
                  cy="19"
                  fill="#FFA726"
                  r="8"
                />
                <path
                  d="M30,36.7c0,0-3.6-6.7-13-6.7S4,36.7,4,36.7V40h26V36.7z"
                  fill="#607D8B"
                />
              </svg>
            </div>
            <ul>
              <li>
                <p
                  class="w-50 font-medium"
                >
                  Aligning your goals with your motives
                </p>
              </li>
              <li>
                <p
                  class="w-50 font-medium"
                >
                  Overcoming decision paralysis
                </p>
              </li>
              <li>
                <p
                  class="w-50 font-medium"
                >
                  Reframing negative self-talk
                </p>
              </li>
              <li>
                <p
                  class="w-50 font-medium"
                >
                  Gaining clarity
                </p>
              </li>
              <li>
                <p
                  class="w-50 font-medium"
                >
                  Challenging yourself
                </p>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </a>
  </li>
</ul>
`;

exports[`CourseList component CourseList snapshots 2`] = `
<nav
  aria-label="Pagination"
  class="flex justify-center items-center text-gray-600 mt-8 lg:mt-0 caret-transparent"
>
  <p
    class="p-2 ml-4 rounded hover:bg-gray-100 cursor-pointer"
  >
    <svg
      class="h-6 w-6"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 19l-7-7 7-7"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </p>
  <p
    class="w-10 h-10 text-center leading-10 rounded hover:bg-gray-100 cursor-pointer bg-gray-200 text-gray-900 font-medium"
  >
    1
  </p>
  <p
    class="w-10 h-10 text-center leading-10 rounded hover:bg-gray-100 cursor-pointer false"
  >
    2
  </p>
  <p
    class="w-10 h-10 text-center leading-10 rounded hover:bg-gray-100 cursor-pointer false"
  >
    3
  </p>
  <p
    class="p-2 ml-4 rounded hover:bg-gray-100 cursor-pointer"
  >
    <svg
      class="h-6 w-6"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 5l7 7-7 7"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </p>
</nav>
`;

exports[`CourseList component List snapshot 1`] = `
<ul
  class="grid gap-5 p-4 justify-items-center tablet:grid-cols-2 laptop:grid-cols-3 desktop:grid-cols-4"
  role="listbox"
>
  <li
    class="block min-w-80 rounded-lg p-3 border shadow-md shadow-gray-500 hover:scale-105"
  >
    <a
      href="course/352be3c6-848b-4c19-9e7d-54fe68fef183"
    >
      <div
        class="relative h-28 w-64 mx-auto group"
        data-testid="group"
      >
        <img
          alt="Lack of Motivation & How to Overcome It"
          class="absolute top-0 h-full w-full rounded-md object-cover group-hover:hidden"
          src="https://wisey.app/assets/images/web/course-covers/lack-of-motivation-how-to-overcome-it/cover.webp"
        />
        <video
          class="absolute top-0 h-full w-full opacity-0 rounded-md object-cover group-hover:opacity-100"
          data-testid="video-player"
          id="352be3c6-848b-4c19-9e7d-54fe68fef183"
          loop=""
          preload="none"
        />
      </div>
      <div
        class="mt-1"
      >
        <p
          class="font-medium"
        >
          Lack of Motivation & How to Overcome It
        </p>
        <p
          class="text-sm text-gray-500"
        >
          Reignite your inner drive by managing factors that dampen your motivation.
        </p>
        <div
          class="flex items-center gap-3 mt-1 text-xs"
        >
          <div>
            <p
              class="text-gray-500"
            >
              Lessons
            </p>
            <p
              class="flex font-medium gap-1"
            >
              <svg
                enable-background="new 0 0 48 48"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                version="1"
                viewBox="0 0 48 48"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="#D1C4E9"
                >
                  <path
                    d="M38,7H10C8.9,7,8,7.9,8,9v6c0,1.1,0.9,2,2,2h28c1.1,0,2-0.9,2-2V9C40,7.9,39.1,7,38,7z"
                  />
                  <path
                    d="M38,19H10c-1.1,0-2,0.9-2,2v6c0,1.1,0.9,2,2,2h28c1.1,0,2-0.9,2-2v-6C40,19.9,39.1,19,38,19z"
                  />
                  <path
                    d="M38,31H10c-1.1,0-2,0.9-2,2v6c0,1.1,0.9,2,2,2h28c1.1,0,2-0.9,2-2v-6C40,31.9,39.1,31,38,31z"
                  />
                </g>
                <circle
                  cx="38"
                  cy="38"
                  fill="#43A047"
                  r="10"
                />
                <polygon
                  fill="#DCEDC8"
                  points="42.5,33.3 36.8,39 34.1,36.3 32,38.5 36.8,43.3 44.6,35.5"
                />
              </svg>
              2
            </p>
          </div>
          <div>
            <p
              class="text-gray-500"
            >
              Rating
            </p>
            <p
              class="flex font-medium gap-1"
            >
              <svg
                enable-background="new 0 0 48 48"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                version="1"
                viewBox="0 0 48 48"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="#00BCD4"
                >
                  <rect
                    height="20"
                    width="10"
                    x="19"
                    y="22"
                  />
                  <rect
                    height="34"
                    width="10"
                    x="32"
                    y="8"
                  />
                  <rect
                    height="12"
                    width="10"
                    x="6"
                    y="30"
                  />
                </g>
                <g
                  fill="#3F51B5"
                >
                  <polygon
                    points="11,8 21,18 21,8"
                  />
                  <rect
                    height="14.1"
                    transform="matrix(-.707 -.707 .707 -.707 10.879 36.506)"
                    width="4"
                    x="11"
                    y="8.9"
                  />
                </g>
              </svg>
              3.5
            </p>
          </div>
          <div>
            <div
              class="flex gap-2"
            >
              <p
                class="text-gray-500"
              >
                Skills
              </p>
              <svg
                enable-background="new 0 0 48 48"
                fill="currentColor"
                height="16"
                stroke="currentColor"
                stroke-width="0"
                version="1"
                viewBox="0 0 48 48"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M40,22h-8l-4,4V12c0-2.2,1.8-4,4-4h8c2.2,0,4,1.8,4,4v6C44,20.2,42.2,22,40,22z"
                  fill="#2196F3"
                />
                <circle
                  cx="17"
                  cy="19"
                  fill="#FFA726"
                  r="8"
                />
                <path
                  d="M30,36.7c0,0-3.6-6.7-13-6.7S4,36.7,4,36.7V40h26V36.7z"
                  fill="#607D8B"
                />
              </svg>
            </div>
            <ul>
              <li>
                <p
                  class="w-50 font-medium"
                >
                  Aligning your goals with your motives
                </p>
              </li>
              <li>
                <p
                  class="w-50 font-medium"
                >
                  Overcoming decision paralysis
                </p>
              </li>
              <li>
                <p
                  class="w-50 font-medium"
                >
                  Reframing negative self-talk
                </p>
              </li>
              <li>
                <p
                  class="w-50 font-medium"
                >
                  Gaining clarity
                </p>
              </li>
              <li>
                <p
                  class="w-50 font-medium"
                >
                  Challenging yourself
                </p>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </a>
  </li>
</ul>
`;

exports[`CourseList component List snapshot 2`] = `
<nav
  aria-label="Pagination"
  class="flex justify-center items-center text-gray-600 mt-8 lg:mt-0 caret-transparent"
>
  <p
    class="p-2 ml-4 rounded hover:bg-gray-100 cursor-pointer"
  >
    <svg
      class="h-6 w-6"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 19l-7-7 7-7"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </p>
  <p
    class="w-10 h-10 text-center leading-10 rounded hover:bg-gray-100 cursor-pointer bg-gray-200 text-gray-900 font-medium"
  >
    1
  </p>
  <p
    class="w-10 h-10 text-center leading-10 rounded hover:bg-gray-100 cursor-pointer false"
  >
    2
  </p>
  <p
    class="w-10 h-10 text-center leading-10 rounded hover:bg-gray-100 cursor-pointer false"
  >
    3
  </p>
  <p
    class="p-2 ml-4 rounded hover:bg-gray-100 cursor-pointer"
  >
    <svg
      class="h-6 w-6"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 5l7 7-7 7"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </p>
</nav>
`;
